packages/cli-e2e/__tests__/angular.specs.ts

```diff
@@ -137,6 +143,26 @@ describe('ui:create:angular', () => {
       await serverProcessManager.killAllProcesses();
     }, 5e3);
 
+    it('should not contain console errors nor warnings', async () => {
+      const interceptedMessages: ConsoleMessage[] = [];
+      const deniedConsoleMessageTypes: ConsoleMessageType[] = [
+        'error',
+        'warning',
+      ];
+
+      page.on('console', (msg) => interceptedMessages.push(msg));
+
+      await page.goto(searchPageEndpoint, {
+        waitUntil: 'networkidle0',
+      });
+
+      expect(
```

Comment:
When it fails, I'd log what is the actual error/warning message that we caught from the console, and log it in the test output.

Will be easier to troubleshoot. Or is it already the case with the way `jest` logs on test failure ?

Reply:
That's what I am trying to do actually.
So far I found 2 quick options (but neither one seems clean):
1. pipe browser process stdout and stderr into process.stdout and process.stderr ([info](https://www.puppeteersharp.com/api/PuppeteerSharp.LaunchOptions.html#PuppeteerSharp_LaunchOptions_DumpIO)).
2. Implement this [workaround](https://github.com/puppeteer/puppeteer/issues/3397#issuecomment-429325514) which use an undocumented method.

The third option is a little bit more complex but more robust
3. Create a CDP session to directly interact with the Chrome DevTools Protocole and create a listener on the [Runtime.consoleAPICalled event](Runtime.consoleAPICalled #)

Reply:
nvm. `Runtime.consoleAPICalled` is awesome!

Reply:
You could probably reuse the Logger to log into files ;)
Like this, if we get a buttload of logs, we can still sieve through them.